core/src/test/java/org/apache/iceberg/view/TestViewVersionParser.java

```diff
@@ -0,0 +1,139 @@
+
+    SQLViewRepresentation firstRepresentation =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from foo")
+            .dialect("spark-sql")
+            .build();
+    SQLViewRepresentation secondRepresentation =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select a, b, c from foo")
+            .dialect("some-sql")
+            .build();
+
+    ViewVersion viewVersion =
+        ImmutableViewVersion.builder()
+            .versionId(1)
+            .timestampMillis(12345)
+            .addRepresentations(firstRepresentation, secondRepresentation)
+            .summary(ImmutableMap.of("operation", "CREATE"))
+            .build();
+
+    String expectedRepresentations =
+        "[{\"type\":\"sql\",\"sql\":\"select * from foo\",\"dialect\":\"spark-sql\"},"
+            + "{\"type\":\"sql\",\"sql\":\"select a, b, c from foo\",\"dialect\":\"some-sql\"}]";
+
+    String expectedViewVersion =
+        String.format(
+            "{\"version-id\":1,\"timestamp-ms\":12345,\"summary\":{\"operation\":\"CREATE\"},\"representations\":%s}",
+            expectedRepresentations);
+
+    Assert.assertEquals(
+        "Should be able to parse valid view version",
+        expectedViewVersion,
+        ViewVersionParser.toJson(viewVersion));
+  }
+
+  @Test
+  public void testFailParsingMissingOperation() {
+    String serializedRepresentations =
+        "[{\"type\":\"sql\",\"sql\":\"select * from foo\",\"dialect\":\"spark-sql\"},"
+            + "{\"type\":\"sql\",\"sql\":\"select a, b, c from foo\",\"dialect\":\"some-sql\"}]";
+
+    String viewVersionMissingOperation =
+        String.format(
+            "{\"version-id\":1,\"timestamp-ms\":12345,\"summary\":{\"some-other-field\":\"some-other-value\"},\"representations\":%s}",
+            serializedRepresentations);
+
+    Assertions.assertThatThrownBy(() -> ViewVersionParser.fromJson(viewVersionMissingOperation))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse summary with missing required field: operation");
+  }
+
+  @Test
+  public void testFailParsingInvalidOperation() {
+    String serializedRepresentations =
+        "[{\"type\":\"sql\",\"sql\":\"select * from foo\",\"dialect\":\"spark-sql\"},"
+            + "{\"type\":\"sql\",\"sql\":\"select a, b, c from foo\",\"dialect\":\"some-sql\"}]";
+
+    String viewVersionMissingOperation =
+        String.format(
+            "{\"version-id\":1,\"timestamp-ms\":12345,\"summary\":{\"operation\":\"unknown-operation\"},\"representations\":%s}",
+            serializedRepresentations);
+
+    Assertions.assertThatThrownBy(() -> ViewVersionParser.fromJson(viewVersionMissingOperation))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse summary with invalid operation: unknown-operation");
+  }
+
+  @Test
+  public void testNullViewVersion() {
+    Assertions.assertThatThrownBy(() -> ViewVersionParser.toJson(null))
```

Comment:
would be good to also add a test with a null string here: `Assertions.assertThatThrownBy(() -> ViewVersionParser.toJson((String) null))`

Reply:
Done! it's a bit awkward since the illegal argument exception thrown is by jackson since it fails in the mapper code itself, but we can at least validate that an exception does get thrown. Also added a test for the null JsonNode case as well